src/dstack/_internal/server/services/fleets.py

```diff
@@ -137,6 +140,27 @@ async def get_plan(
 ) -> FleetPlan:
     # TODO: refactor offers logic into a separate module to avoid depending on runs
 
+    if spec.configuration.ssh_config and spec.configuration.ssh_config.hosts:
```

Comment:
The entire check should be moved into a separate function like `_check_ssh_hosts_not_yet_added()`